fil_deal_stats.go

```diff
@@ -0,0 +1,75 @@
+package main
+
+import (
+	"context"
+	"sync"
+)
+
+type dealStats struct {
+	hf                     *heyFil
+	refreshLock            sync.RWMutex
+	dealCountByParticipant map[string]int64
+	latestRefreshErr       error
+}
+
+func (ds *dealStats) start(ctx context.Context) {
+	go func() {
+		for {
+			ds.latestRefreshErr = ds.refresh(ctx)
+			select {
+			case <-ctx.Done():
+				return
+			case <-ds.hf.dealStatsRefreshInterval.C:
+				ds.latestRefreshErr = ds.refresh(ctx)
+			}
+		}
+	}()
+}
+
+func (ds *dealStats) refresh(ctx context.Context) error {
+	ch, err := ds.hf.chainHead(ctx)
+	if err != nil {
+		return err
+	}
+	epoch := ch.Height
+
+	deals, err := ds.hf.stateMarketDeals(ctx)
+	if err != nil {
+		return err
+	}
+
+	dealCountByParticipant := make(map[string]int64)
+	var totalDealCount int64
+	for {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case dr, ok := <-deals:
+			if !ok {
+				ds.refreshLock.Lock()
+				ds.dealCountByParticipant = dealCountByParticipant
+				ds.refreshLock.Unlock()
+				ds.hf.metrics.notifyDealCount(totalDealCount)
+				logger.Infow("fetched state market deals", "count", totalDealCount)
```

Comment:
rather than just doing dealCount - it would be interesting if we could try correlate on contextID since we know filecoin providers will put in their deal ID as the context ID

Reply:
Sounds good. 
What's here is equivalent to how we measure it in index-observer. 

Ok to do this in a separate PR?

Reply:
for sure :+1: